include/net/net_if.h

```diff
@@ -271,6 +274,28 @@ struct net_if {
 	} ipv4;
 #endif /* CONFIG_NET_IPV4 */
 
+#if defined(CONFIG_NET_IPV4_AUTOCONF)
+    struct {
+        /** Number of sent probe requests */
+        uint8_t probe_cnt;
+        /** Number of sent announcements */
+        uint8_t announce_cnt;
+        /** Incoming conflict count */
+        uint8_t conflict_cnt;
+        /** Requested IP addr */
+        struct in_addr current_ip;
+        /** Requested IP addr */
+        struct in_addr requested_ip;
+        /**
+        *  IPV4 Autoconf state in the process of network
+        *  address allocation.
+        */
+        enum net_ipv4_autoconf_state state;
+    } ipv4_autoconf;
```

Comment:
I think you can move ipv4_autoconf into the above ipv4 struct. This is part of ipv4.